docs/howto/upgrade-cluster/aws.md

```diff
@@ -134,62 +165,86 @@ cluster is unused or that the maintenance is communicated ahead of time.
+### 5. Upgrade node groups up to three minor versions until it matches the version of the k8s control plane
+
+   ```{important}
+   A node's k8s software (`kubelet`) can be up to three minor versions
+   **behind** the control plane version [^2] if kublet is at least at version 1.25.
+   Due to this, you can plan your cluster upgrade to only involve the minimum
+   number of node group upgrades.
 
-   A node's k8s software (`kubelet`) can be up to two minor versions ahead or
-   behind the control plane version.[^1] Due to this, you can plan your cluster
-   upgrade to only involve one node group upgrade even if you increment the
-   control plane four minor versions.
+   So if you upgrade from k8s 1.25 to 1.28, you can for example upgrade the k8s
+   control plane  three steps in a row, from 1.25 to 1.26, then from 1.26
+   to 1.27 and then from 1.27 to 1.28. This way, the node groups were left
+   behind the control plane by three minor versions, which is acceptable.
 
-   So if you upgrade from k8s 1.21 to 1.24, you can for example upgrade the k8s
-   control plane from 1.21 to 1.22, then upgrade the node groups from 1.21 to
-   1.24, followed by upgrading the control plane two steps in a row.
+   Then, you can upgrade the node groups from 1.25 to 1.28.
+   ```
 
-   To upgrade (unmanaged) node groups, you delete them and then them back. When
+   To upgrade (unmanaged) node groups, you delete them and then add them back in. When
    adding them back, make sure your cluster config's k8s version is what you
    want the node groups to be added back as.
 
-   1. Update the k8s version in the config temporarily
+   #### 5.1. Update the k8s version in the config temporarily if needed
 
       This is to influence the k8s software version for the nodegroup's we
-      create only. We can choose something two minor versions of the current k8s
+      create only. We cannot choose a version ahead of the current k8s
       control plane version.
 
-   2. Add a new core node group (like `core-b`)
+      Make sure the cluster's jsonnet config's version field matches the
+      current version of the control plane and this version is **no more
+      than three minor versions** ahead of the node groups versions.
 
-      Rename (part 1/3) the config file's entry for the core node group
-      temporarily when running this command, either from `core-a` to `core-b` or
-      the other way around.
+   #### 5.2. Rename part 1: add a new core node group (like `core-b`)
+
+      Rename the `CLUSTER_NAME.jsonnet` config file's entry for the core node
+      group temporarily when running this command, either from `core-a` to `core-b` or
+      the other way around, then regenerate the `$CLUSTER_NAME.eksctl.yaml` file and
+      then create the new nodegroup.
+
+      ```bash
+      # re-generate an eksctl config file for use with eksctl
+      jsonnet $CLUSTER_NAME.jsonnet > $CLUSTER_NAME.eksctl.yaml
+      ```
```

Comment:
I've appreciated doing temporary changes directly from the generated .yaml file instead of re-generating it repeatedly. Consider if you think that is an easier or more complicated strategy to use and document.

Reply:
Yes, yes! Agree! I actually did the same, but felt the need to run a diff to make sure I got the node pool name right and I was not changing something else by mistake.

Hmm, instead, maybe temporarily git adding the eksctl.yaml file in the beginning might be more straightforward, as then you get to have the diff. Wdyt?

Reply:
@consideRatio, just realized you had this written all along: _If you update the .jsonnet file, make sure to re-generate the .yaml file
   before using `eksctl`. Respectively if you update the .yaml file directly,
   remember to update the .jsonnet file._

Reply:
Addressed this in https://github.com/2i2c-org/infrastructure/pull/3420/commits/5684ef9a4c8f71fcd3a0b3913879d22329540b66